osu.Game/Overlays/MusicController.cs

```diff
@@ -385,15 +387,18 @@ public bool AllowTrackAdjustments
         /// </remarks>
         public void ResetTrackAdjustments()
         {
+            // todo: we probably want a helper method rather than this.
             CurrentTrack.RemoveAllAdjustments(AdjustableProperty.Balance);
             CurrentTrack.RemoveAllAdjustments(AdjustableProperty.Frequency);
             CurrentTrack.RemoveAllAdjustments(AdjustableProperty.Tempo);
             CurrentTrack.RemoveAllAdjustments(AdjustableProperty.Volume);
 
             if (allowTrackAdjustments)
             {
+                CurrentTrack.BindAdjustments(modTrackAdjustments = new AudioAdjustments());
+
                 foreach (var mod in mods.Value.OfType<IApplicableToTrack>())
-                    mod.ApplyToTrack(CurrentTrack);
+                    mod.ApplyToTrack(modTrackAdjustments);
```

Comment:
It feels like a half-solution when the OP is saying the changes in #18196 feel better (referring to applying via bindables/not storing the "track") but this is still feeding into that.

Have you considered changing these methods to `CreateTrackAdjustments`/`CreateSampleAdjustments()` instead, to make the mods return `AudioAdjustment` objects?

Reply:
The adjustments that music controller feeds to mods are thrown away on every adjustment reset. So it no longer matters what the mods do, because they essentially can only store temporary adjustments even if they try to store them.